### ISSUE TITLE
Understanding the -slim node_modules option in issue 101

### ISSUE THREAD

**Issue by christertamm** [jxcore/jxcore#384]
I try to package my program, named "server.js" like this:

demo:/opt/TVI/nodesockets # ../bin/jx package server.js
Processing the folder..
skipping existing package server.jx
JXP project file (server.jxp) is ready.

preparing the JX file..
Compiling server 1.0
adding script misc.js
adding script node_modules/libxmljs/examples/cdata.js
adding script node_modules/libxmljs/index.js
adding script node_modules/libxmljs/lib/bindings.js
[...]
and two thousand lines later:

adding asset node_modules/ws/node_modules/utf-8-validate/node_modules/nan/nan_string_bytes.h
adding asset node_modules/ws/node_modules/utf-8-validate/src/validation.cc
adding asset server.jxp
Warning! ....idate/build/Release/validation.node
Adding a .node (native) file into package may fail the application
Check the related discussion from https://github.com/jxcore/jxcore/issues/101
[OK] compiled file is ready (server.jx)
demo:/opt/TVI/nodesockets #

Seems like the file "validation.node" might cause some problems. Well, let's try and see what happens:

demo:/opt/TVI/nodesockets # ../bin/jx server.jx

/opt/TVI/nodesockets/node_modules/libxmljs/lib/element.js.jx:25
Element.prototype = bindings.Element.prototype;
                                    ^
TypeError: Cannot read property 'prototype' of undefined
    at Object.<anonymous> (/opt/TVI/nodesockets/node_modules/libxmljs/lib/element.js.jx:25:37)
    at Module._compile (module.js:519:26)
    at Object.Module._extensions..jx (module.js:1641:7)
    at Module.load (module.js:346:36)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:378:17)
    at require (module.js:396:17)
    at Object.<anonymous> (/opt/TVI/nodesockets/node_modules/libxmljs/lib/document.js.jx:4:15)
    at Module._compile (module.js:519:26)
    at Object.Module._extensions..jx (module.js:1641:7)
demo:/opt/TVI/nodesockets #

OK, time to check the discussion at https://github.com/jxcore/jxcore/issues/101:

obastemur's solution seems promising, so I try it: 

demo:/opt/TVI/nodesockets # ../bin/jx package server.js -slim node_modules
Processing the folder..
skipping existing package server.jx
JXP project file (server.jxp) is ready.

preparing the JX file..
Compiling server 1.0
adding script misc.js
adding script server.js
adding asset .DS_Store
[OK] compiled file is ready (server.jx)

As I understand it, what I have here is just misc.js and server.js packaged. If I leave everything as it is, this package SHOULD be able to run just fine, right? It should find the node_modules folder and use the modules in it, just like node.js does when it runs server.js?

demo:/opt/TVI/nodesockets # ../bin/jx server.jx

/opt/TVI/nodesockets/node_modules/libxmljs/node_modules/bindings/bindings.js:83
        throw e
              ^
Error: /opt/TVI/nodesockets/node_modules/libxmljs/build/Release/xmljs.node: undefined symbol: node_module_register
    at Object.process.dlopen [as .node] (node.js:670:25)
    at Module.load (module.js:346:36)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:378:17)
    at require (module.js:396:17)
    at bindings (/opt/TVI/nodesockets/node_modules/libxmljs/node_modules/bindings/bindings.js:76:44)
    at Object.<anonymous> (/opt/TVI/nodesockets/node_modules/libxmljs/lib/bindings.js:1:133)
    at Module._compile (module.js:519:26)
    at Object.Module._extensions..js (module.js:549:10)
    at Module.load (module.js:346:36)

What am I doing wrong? Or have I perhaps misunderstood this completely?

**Comment by ktrzeciaknubisa** [comment]
Hi @christertamm I think, you get it all right. Few questions for you though:

1. when you run `jx server.jx` (the one with slimed node_modules) - is this the same machine where you created the package, and node_modules were installed there too? 
2. Does the app run unpackaged with `jx server.js` ? 
3. Did you install node_modules with `npm install` or `jx install`?

**Comment by christertamm** [comment]
1. Yes, it is the same machine where I created the package. For now, that is. If I get this working, I intend to deploy my app on other machines, running the same OS, applications, etc. But first things first; I have only tried this on the same machine so far.

2. I haven't tested that yet, I will have to try it and see what happens and return to you.
3. node_modules are installed with "npm install". Would it make a difference if i used "jx install" instead?
